### PR TITLE
[PRISM] Support NO_COLOR environment variable

### PR DESCRIPTION
**Pull request by eightbitraptor**
Fixes [Bug #20940]

### CODE REVIEW

**Review by kddnewton — APPROVED** [review]
This works though the British spelling threw me haha

**Comment by eightbitraptor** [comment]
> This works though the British spelling threw me haha

Oh, you mean in the comment? Damn, I tried so hard to be 🇺🇸 ! 

I'll fix that

**Comment by kddnewton** [comment]
I meant in the function name actually!

File: prism_compile.c [hunk 1]
```diff
@@ -10042,6 +10042,17 @@ pm_iseq_pre_execution_p(rb_iseq_t *iseq)
     }
 }
 
+/**
+ * Colorise the output if the terminal is a TTY and the `NO_COLOR` environment
+ * variable is not set.
+ */
+int
```
Comment (XrXr) [inline prism_compile.c:10049]:
```suggestion
static int
```

File: prism_compile.c [hunk 2]
```diff
@@ -10537,7 +10548,7 @@ pm_parse_process_error(const pm_parse_result_t *result)
                 pm_list_node_t *list_node = (pm_list_node_t *) error;
                 pm_list_t error_list = { .size = 1, .head = list_node, .tail = list_node };
 
-                pm_parse_errors_format(parser, &error_list, &buffer, rb_stderr_tty_p(), false);
+                pm_parse_errors_format(parser, &error_list, &buffer, rb_stderr_colorize_p(), false);
```
Comment (nobu) [inline prism_compile.c:10551]: Why checking the env repeatedly?

**Comment by nobu** [comment]
I think it is possible to use highlight even without colors, on a tty.

**Comment by nobu** [comment]
#12329

**Comment by eightbitraptor** [comment]
Closing in favour of https://github.com/ruby/ruby/pull/12329

**Pull request closed without merging**